core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -69,24 +69,46 @@ class TopicConfigHandler(private val replicaManager: ReplicaManager,
 
     val logs = logManager.logsByTopic(topic)
     val wasRemoteLogEnabledBeforeUpdate = logs.exists(_.remoteLogEnabled())
+    var oldLogPolicy = logs.head.config.remoteLogDisablePolicy()
+    if (oldLogPolicy == null)
+      oldLogPolicy = "retain"
 
     logManager.updateTopicConfig(topic, props, kafkaConfig.remoteLogManagerConfig.isRemoteStorageSystemEnabled())
-    maybeBootstrapRemoteLogComponents(topic, logs, wasRemoteLogEnabledBeforeUpdate)
+    maybeBootstrapRemoteLogComponents(topic, logs, wasRemoteLogEnabledBeforeUpdate, oldLogPolicy)
   }
 
   private[server] def maybeBootstrapRemoteLogComponents(topic: String,
                                                         logs: Seq[UnifiedLog],
-                                                        wasRemoteLogEnabledBeforeUpdate: Boolean): Unit = {
+                                                        wasRemoteLogEnabledBeforeUpdate: Boolean,
+                                                        oldLogPolicy: String): Unit = {
     val isRemoteLogEnabled = logs.exists(_.remoteLogEnabled())
+    var newRemoteLogPolicy = logs.head.config.remoteLogDisablePolicy()
+    if (newRemoteLogPolicy == null)
+      newRemoteLogPolicy = "retain"
+
+
+
+    val (leaderPartitions, followerPartitions) =
+      logs.flatMap(log => replicaManager.onlinePartition(log.topicPartition)).partition(_.isLeader)
     // Topic configs gets updated incrementally. This check is added to prevent redundant updates.
     if (!wasRemoteLogEnabledBeforeUpdate && isRemoteLogEnabled) {
-      val (leaderPartitions, followerPartitions) =
-        logs.flatMap(log => replicaManager.onlinePartition(log.topicPartition)).partition(_.isLeader)
       val topicIds = Collections.singletonMap(topic, replicaManager.metadataCache.getTopicId(topic))
       replicaManager.remoteLogManager.foreach(rlm =>
         rlm.onLeadershipChange(leaderPartitions.toSet.asJava, followerPartitions.toSet.asJava, topicIds))
-    } else if (wasRemoteLogEnabledBeforeUpdate && !isRemoteLogEnabled) {
-      warn(s"Disabling remote log on the topic: $topic is not supported.")
+    }
+
+    // When there's a configRecord related to topic, we'll invoke updateLogConfig and enter here.
+    // So, here, we can check if the tiered storage is enabled or disabled, or remote log policy change or not, then do stopPartitions
+    // for each partition.
+    if (wasRemoteLogEnabledBeforeUpdate != isRemoteLogEnabled || !oldLogPolicy.equals(newRemoteLogPolicy)) {
+      val stopPartitions: java.util.HashSet[StopPartition] = new java.util.HashSet[StopPartition]()
+      leaderPartitions.toSet.asJava.stream().forEach(partition => {
```

Comment:
```suggestion
      leaderPartitions.foreach { partition =>
```

Reply:
We also have to stop the follower RLMM tasks: (copied the below snipped from our internal branch)

```
logs.map(log => {
        log.maybeIncrementLogStartOffsetAsRemoteLogStorageDisabled()
        log.topicPartition
      })
        .groupBy(tp => replicaManager.onlinePartition(tp).exists(_.isLeader))
        .foreach {
          case (isLeader, partitions) =>
            val stopPartitions = partitions
              .map(partition => StopPartition(partition, deleteLocalLog = false, deleteRemoteLog = isLeader, stopRLMM = true))
              .toSet
            replicaManager.remoteLogManager.foreach { rlm =>
              rlm.stopPartitions(stopPartitions, (_, _) => {})
            }
        }
    }
```

Reply:
Hmm... good point, let me think about it.

Reply:
I think we need to stop RLMM only when deletePolicy is `delete`. For `retain`, we still need RLMM because we need to get remote log metadata to do the retention check. Will work on that.